cupyx/scipy/linalg/special_matrices.py

```diff
@@ -0,0 +1,552 @@
+    return H if full else H[1:]
+
+
+def hilbert(n):
+    """Create a Hilbert matrix of order ``n``.
+
+    Returns the ``n`` by ``n`` array with entries ``h[i,j] = 1 / (i + j + 1)``.
+
+    Args:
+        n (int): The size of the array to create.
+
+    Returns:
+        cupy.ndarray: The Hilbert matrix.
+
+    .. seealso:: :func:`scipy.linalg.hilbert`
+    """
+    values = cupy.arange(1, 2*n, dtype=float)
+    cupy.divide(1.0, values, values)
+    return hankel(values[:n], r=values[n-1:])
+
+
+# TODO: invhilbert(n, exact=False)
+# TODO: pascal(n, kind='symmetric', exact=True)
+# TODO: invpascal(n, kind='symmetric', exact=True)
+
+
+def dft(n, scale=None):
+    """Discrete Fourier transform matrix.
+
+    Create the matrix that computes the discrete Fourier transform of a
+    sequence. The nth primitive root of unity used to generate the matrix is
+    exp(-2*pi*i/n), where i = sqrt(-1).
+
+    Args:
+        n (int): Size the matrix to create.
+        scale (str, optional): Must be None, 'sqrtn', or 'n'.
+            If ``scale`` is 'sqrtn', the matrix is divided by ``sqrt(n)``.
+            If ``scale`` is 'n', the matrix is divided by ``n``.
+            If ``scale`` is None (default), the matrix is not normalized, and
+            the return value is simply the Vandermonde matrix of the roots of
+            unity.
+
+    Returns:
+        (cupy.ndarray): The DFT matrix.
+
+    Notes:
+        When ``scale`` is None, multiplying a vector by the matrix returned by
+        ``dft`` is mathematically equivalent to (but much less efficient than)
+        the calculation performed by ``scipy.fft.fft``.
+
+    .. seealso:: :func:`scipy.linalg.dft`
+    """
+    if scale not in (None, 'sqrtn', 'n'):
+        raise ValueError("scale must be None, 'sqrtn', or 'n'; "
+                         "%r is not valid." % (scale,))
+    r = cupy.arange(n, dtype='complex128')
+    r *= -2j*cupy.pi/n
+    omegas = cupy.exp(r, out=r)[:, None]
+    m = omegas ** cupy.arange(n)
+    if scale is not None:
+        m *= (1/math.sqrt(n)) if scale == 'sqrtn' else (1/n)
+    return m
+
+
+def fiedler(a):
+    """Returns a symmetric Fiedler matrix
+
+    Given an sequence of numbers ``a``, Fiedler matrices have the structure
+    ``F[i, j] = np.abs(a[i] - a[j])``, and hence zero diagonals and nonnegative
+    entries. A Fiedler matrix has a dominant positive eigenvalue and other
+    eigenvalues are negative. Although not valid generally, for certain inputs,
+    the inverse and the determinant can be derived explicitly.
+
+    Args:
+        a (cupy.ndarray): coefficient array
+
+    Returns:
+        cupy.ndarray: the symmetric Fiedler matrix
+
+    .. seealso:: :func:`cupyx.scipy.linalg.circulant`
+    .. seealso:: :func:`cupyx.scipy.linalg.toeplitz`
+    .. seealso:: :func:`scipy.linalg.fiedler`
+    """
+    if a.ndim != 1:
+        raise ValueError("Input 'a' must be a 1D array.")
+    if a.size == 0:
+        return cupy.zeros(())
```

Comment:
Please add a test for this case.

Reply:
Added and fixed this line to return the correct value.